src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/PlatformManager.java

```diff
@@ -0,0 +1,240 @@
+            if (!platform.getPayment().isPresent())
+                rec.setNeedsEmail(true);
+            /* if platform cannot identify worker, we need to do that with a email adress */
+            if (!platform.getWorker().isPresent())
+                rec.setNeedsEmail(true);
+
+            rec.setRenderCalibrations(platform.isCalibrationAllowed());
+
+            platformOps.createPlatform(rec);
+        });
+    }
+
+    /**
+     * Will get you the instance of a platform interface of a platform, this instance is the same for all calls
+     * @param name The name of the instance to use
+     * @return The optional crowd platform instance
+     */
+    public Optional<Platform> getPlatform(String name) {
+        return Optional.ofNullable(platforms.get(name));
+    }
+
+    /**
+     * Will return the Worker interface which should be used to identify workers for the given platform
+     *
+     * @param name The name of the platform
+     * @return The interface used to identify a worker
+     */
+    public Worker getWorker(String name) {
+        return getPlatform(name)
+                .orElseThrow(() -> new IllegalArgumentException("Platform not found"))
+                .getWorker().orElse(fallbackWorker);
+    }
+
+    /**
+     * Will return the payment service which should be used for the given platform
+     * If there is no Platform with the given name None is returned.
+     *
+     * @param name The name of the platform to use
+     * @return The interface used for payment
+     */
+    public Payment getPlatformPayment(String name) {
+        return getPlatform(name)
+                .orElseThrow(() -> new IllegalArgumentException("Platform not found"))
+                .getPayment().orElse(fallbackPayment);
+    }
+
+    /**
+     * Publish the given experiment on the platform.
+     * The method will update the database with the new public task
+     *
+     * @param name The name of the platform
+     * @param experiment The experiment to publish
+     * @return None if the platform does not exist
+     */
+    public CompletableFuture<Boolean> publishTask(String name, Experiment experiment) throws TaskOperationException {
+
+        TaskRecord record = new TaskRecord();
+        record.setExperiment(experiment.getId());
+        record.setStatus(TaskStatus.running);
+        record.setCrowdPlatform(name);
+
+        TaskRecord result = tasksOps.createTask(record);
+        if (result == null)
+            throw new TaskOperationException("Task could not be created");
+
+        return getPlatform(name)
+                .map(platform1 -> platform1.publishTask(experiment))
+                .orElseThrow(() -> new IllegalArgumentException("Platform not found!"))
+                .handle((s1, throwable) -> {
+                    if (s1 != null) {
+                        result.setPlatformData(s1);
+                    } else {
+                        result.setStatus(TaskStatus.stopped);
```

Comment:
you have to call method in the operations package. This will change nothing in the database and i don`t guarantee that .store will work